arch/Kconfig

```diff
@@ -805,6 +805,7 @@ config CFI_CLANG
 	bool "Use Clang's Control Flow Integrity (CFI)"
 	depends on ARCH_SUPPORTS_CFI_CLANG
 	depends on $(cc-option,-fsanitize=kcfi)
+	depends on !RUST || $(cc-option,-fsanitize=kcfi,-fsanitize-cfi-icall-experimental-normalize-integers)
```

Comment:
Should we move this to the `RUST` symbol? So far we have kept constraints on the `RUST` config symbol, because Rust may be seen as the "additional feature" and it was useful to see what was needed. I think eventually it will make more sense the other way around, i.e. like here, so that Rust is the "base feature".